src/Graphics/Glyphs/GlyphGeom.cc

```diff
@@ -42,167 +43,150 @@ GlyphGeom::GlyphGeom() : numVBOElements_(0), lineIndex_(0)
-  vboSize += static_cast<uint32_t>(normals_.size()) * 3 * sizeof(float);
-  if (colorScheme == ColorScheme::COLOR_IN_SITU || colorScheme == ColorScheme::COLOR_MAP)
-    vboSize += static_cast<uint32_t>(colors_.size()) * 4 * sizeof(float); //RGBA
-  iboSize = static_cast<uint32_t>(indices_.size()) * sizeof(uint32_t);
-  /// \todo To reduce memory requirements, we can use a 16bit index buffer.
-
-  /// \todo To further reduce a large amount of memory, get rid of the index
-  ///       buffer and use glDrawArrays to render without an IBO. An IBO is
-  ///       a waste of space.
-  ///       http://www.opengl.org/sdk/docs/man3/xhtml/glDrawArrays.xml
+  if (isTransparent) uniforms.push_back(SpireSubPass::Uniform("uTransparency", static_cast<float>(transparencyValue)));
 
-  /// \todo Switch to unique_ptrs and move semantics.
-  std::shared_ptr<spire::VarBuffer> iboBufferSPtr(new spire::VarBuffer(iboSize));
-  std::shared_ptr<spire::VarBuffer> vboBufferSPtr(new spire::VarBuffer(vboSize));
-
-  // Accessing the pointers like this is contrived. We only do this for
-  // speed since we will be using the pointers in a tight inner loop.
-  auto iboBuffer = iboBufferSPtr.get();
-  auto vboBuffer = vboBufferSPtr.get();
-
-  //write to the IBO/VBOs
-
-  for (auto a : indices_)
-    iboBuffer->write(a);
-
-  BBox newBBox;
-
-  const bool writeNormals = normals_.size() == points_.size();
-  for (size_t i = 0; i < points_.size(); i++)
+  size_t pointsLeft = points_.size();
+  size_t startOfPass = 0;
+  int passNumber = 0;
+  while(pointsLeft > 0)
   {
-    // Write first point on line
-    vboBuffer->write(static_cast<float>(points_.at(i).x()));
-    vboBuffer->write(static_cast<float>(points_.at(i).y()));
-    vboBuffer->write(static_cast<float>(points_.at(i).z()));
+    std::string passID = uniqueNodeID + "_" + std::to_string(passNumber++);
+    std::string vboName = passID + "VBO";
+    std::string iboName = passID + "IBO";
+    std::string passName = passID + "Pass";
+
+    const static size_t maxPointsPerPass = 3 << 24; //must be a numebr divisible by 2, 3 and, 4
+    uint32_t pointsInThisPass = std::min(pointsLeft, maxPointsPerPass);
+    size_t endOfPass = startOfPass + pointsInThisPass;
+    pointsLeft -= pointsInThisPass;
+
+    size_t vboSize = static_cast<size_t>(pointsInThisPass) * numAttributes * sizeof(float);
+    size_t iboSize = static_cast<size_t>(pointsInThisPass) * sizeof(uint32_t);
+    std::shared_ptr<spire::VarBuffer> iboBufferSPtr(new spire::VarBuffer(iboSize));
+    std::shared_ptr<spire::VarBuffer> vboBufferSPtr(new spire::VarBuffer(vboSize));
+    auto iboBuffer = iboBufferSPtr.get();
+    auto vboBuffer = vboBufferSPtr.get();
+
+    for (auto a : indices_) if(a >= startOfPass && a < endOfPass)
+      iboBuffer->write(static_cast<uint32_t>(a - startOfPass));
+
+    BBox newBBox;
+    for (size_t i = startOfPass; i < endOfPass; ++i)
+    {
+      newBBox.extend(Point(points_.at(i).x(), points_.at(i).y(), points_.at(i).z()));
```

Comment:
Might be better to grab each `ith` point/normal out of the loop, rather than call slow `at` multiple times:
```
const auto& point = points_.at(i);
...
const auto& normal = normals_.at(i);
```